invenio_files_rest/models.py

```diff
@@ -946,11 +972,18 @@ def copy(self, bucket=None, key=None):
             Default: current object key.
         :returns: The copied object version.
         """
-        return ObjectVersion.create(
-            self.bucket if bucket is None else as_bucket(bucket),
-            key or self.key,
-            _file_id=self.file_id
-        )
+        with db.session.begin_nested():
+            o = ObjectVersion.create(
+                self.bucket if bucket is None else as_bucket(bucket),
+                key or self.key,
+                _file_id=self.file_id
+            )
+            db.session.add(o)
```

Comment:
``ObjectVersion.create`` is already running in a nested transaction so it should not be necessary do it here as well (i.e. remove ``with db.session.begin_nested()`` + ``db.session.add(o)``.